crates/biome_js_analyze/src/lint/nursery/use_readonly_class_properties.rs

```diff
@@ -0,0 +1,799 @@
+                if is_mutable_and_optionally_private(&ts_property, private_only) =>
+            {
+                Some(PropOrParam::TsPropertyParameter(ts_property))
+            }
+            _ => None,
+        })
+        .collect()
+}
+
+/// Collects all property reassignments within class method members or setters,
+/// including assignments and updates to this or static properties.
+fn collect_all_class_property_updates(members: &JsClassMemberList) -> Vec<Text> {
+    members
+        .iter()
+        .filter_map(|member| match member {
+            AnyJsClassMember::JsMethodClassMember(method) => {
+                if let Ok(body) = method.body() {
+                    let this_aliases = collect_class_member_props_mutations(&body);
+                    Some(
+                        find_all_assignment_names(method.syntax(), &this_aliases)
+                            .collect::<Vec<_>>()
+                            .into_iter(),
+                    )
+                } else {
+                    None
+                }
+            }
+            AnyJsClassMember::JsSetterClassMember(setter) => {
+                if let Ok(body) = setter.body() {
+                    let this_aliases = collect_class_member_props_mutations(&body);
+                    Some(
+                        find_all_assignment_names(setter.syntax(), &this_aliases)
+                            .collect::<Vec<_>>()
+                            .into_iter(),
+                    )
+                } else {
+                    None
+                }
+            }
+            AnyJsClassMember::JsConstructorClassMember(constructor) => {
+                if let Ok(body) = constructor.body() {
+                    Some(collect_class_member_props_mutations(&body).into_iter())
+                } else {
+                    None
+                }
+            }
+            _ => None,
+        })
+        .flatten()
+        .collect::<Vec<_>>()
+}
+
+/// Checks recursively the assignment operand equals a reference to `this` (e.g., `this.privateProp`)
+fn contains_this_or_static_member_kind(
+    assignment: &JsStaticMemberAssignment,
+    this_aliases: &[Text],
+) -> bool {
+    if let Ok(object) = assignment.object() {
+        let js_this_assignment = object.as_js_this_expression();
+
+        if js_this_assignment.is_some() {
+            return true;
+        }
+
+        if let Some(js_identifier_expression) = object.as_js_identifier_expression() {
+            if let Ok(name) = js_identifier_expression.name() {
+                return this_aliases
+                    .iter()
+                    .any(|alias| alias.eq(&name.to_trimmed_text()));
+            }
+        }
+    }
+
+    false
+}
+
+/// Collects recursively all assignment expressions and update expressions (e.g., = or ++) within a given syntax node.
+fn find_all_assignment_names(
+    node: &SyntaxNode<JsLanguage>,
+    this_aliases: &Vec<Text>,
```

Comment:
It's more idiomatic to pass a slice `&[Text]`. You can retrieve a slice from a vector using `.as_slice()`